### ISSUE TITLE
void TextureUnitState::setNumMipmaps(int numMipmaps) nullPtr

### ISSUE THREAD

**Issue by arkeon7** [OGRECave/ogre#733]
#### System Information
- Ogre Version: master
- Operating System / Platform: Windows
- RenderSystem: OpenGL

#### Detailled description
When I create a texture like this:
```
Ogre::TextureUnitState* shadow = nPass->createTextureUnitState();
  shadow->setName("SO3/shadows/LISPSM/0");
  shadow->setContentType(Ogre::TextureUnitState::CONTENT_SHADOW);
  shadow->setTextureAddressingMode(Ogre::TextureUnitState::TAM_BORDER);
  shadow->setTextureBorderColour(Ogre::ColourValue(1.0f, 1.0f, 1.0f, 1.0f));
  shadow->setNumMipmaps(0);
```

void TextureUnitState::setNumMipmaps(int numMipmaps)
do not test if the frame is null or not, this result to a crash.

Correction:
`
void TextureUnitState::setNumMipmaps(int numMipmaps)
    {
        for (auto& frame : mFramePtrs)
          if (frame)
          {
            frame->setNumMipmaps(numMipmaps == MIP_DEFAULT
              ? TextureManager::getSingleton().getDefaultNumMipmaps()
              : numMipmaps);
          }
    }
`

**Comment by paroj** [comment]
thanks. The bug however is in setContentType which allocates null textures. Also you probably mean set**Texture**Name

**Comment by arkeon7** [comment]
* Also you probably mean setTextureName
No I just set the texture unit name for debug purpose.
The texture ptr was managed with shadows from Ogre.
Well I'm not sure what is broken yet, making an empty texture and set it afterward was possible before I think.

**Comment by paroj** [comment]
the fix for #729 was that `TextureUnit` no longer caches any `Texture` state. If there is no Texture assigned, calls to e.g. `TextureUnitState::setNumMipmaps` are without effect.

With this change, setContentType still had some unneeded side-effect.
